xds/src/main/java/io/grpc/xds/sds/trust/SdsTrustManagerFactory.java

```diff
@@ -47,19 +46,17 @@
   private static final Logger logger = Logger.getLogger(SdsTrustManagerFactory.class.getName());
   private SdsX509TrustManager sdsX509TrustManager;
 
-  /** Constructor that loads certs from a file. */
-  public SdsTrustManagerFactory(
-      String certsFile, @Nullable CertificateValidationContext certContext)
+  /** Constructor constructs from a {@link CertificateValidationContext}. */
+  public SdsTrustManagerFactory(CertificateValidationContext certContext)
       throws CertificateException, IOException, CertStoreException {
-    this(CertificateUtils.toX509Certificates(new File(certsFile)), certContext);
+    checkNotNull(certContext, "certContext");
+    String certsFile = getTrustedCaFromCertContext(certContext);
```

Comment:
this certsFile can be null but nothing is handling null case

Reply:
If you see how `SdsTrustManagerFactory(CertificateValidationContext)` is called from `SslContextSecretVolumeSecretProvider.buildSslContextFromSecrets` after going thru various `validate*` methods you will see that if `certContext` is non-null it has to have the file-name (non-null and file-name type data-source). That's why I didn't add that check but it is easy to add one if you insist.